### PR TITLE
migrate lifecyclePolicy.updateOnRepair to GA

### PR DESCRIPTION
**Pull request by askubis**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15244

Add support for forcing update on repair in instance lifecycle policy in compute_instance_group_manager and compute_region_instance_group_manager resources (GA).

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- While running the tests one of them fails (TestAccInstanceGroupManager_waitForStatus), but it does not use the new field and it simply times out.
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: enhancement
compute: added field `instance_lifecycle_policy` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` (ga)
```

### CODE REVIEW

**Review by NickElliot — COMMENTED** [review]
Looks like a rebase is needed -- there appears to be a problem with the GA so hopefully it resolves with that.

**Review by NickElliot — COMMENTED** [review]
I'm not referring to the red test but the following error on build: 

```nickelliot-macbookpro:terraform-provider-google nickelliot$ make
sh -c "'/Users/nickelliot/go/src/github.com/hashicorp/terraform-provider-google/scripts/gofmtcheck.sh'"
==> Checking that code complies with gofmt requirements...
go vet
go: downloading golang.org/x/oauth2 v0.9.0
go: downloading google.golang.org/api v0.130.0
go: downloading google.golang.org/grpc v1.56.1
go: downloading golang.org/x/net v0.12.0
go: downloading cloud.google.com/go/bigtable v1.19.0
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20230629202037-9506855d4529
go: downloading google.golang.org/protobuf v1.31.0
go: downloading golang.org/x/sys v0.10.0
go: downloading github.com/googleapis/gax-go/v2 v2.11.0
go: downloading cloud.google.com/go v0.110.2
go: downloading cloud.google.com/go/iam v1.1.0
go: downloading cloud.google.com/go/longrunning v0.5.0
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.2.5
go: downloading github.com/envoyproxy/go-control-plane v0.11.1-0.20230524094728-9239064ad72f
go: downloading golang.org/x/crypto v0.11.0
go: downloading github.com/cespare/xxhash/v2 v2.2.0
go: downloading github.com/cncf/xds/go v0.0.0-20230607035331-e9ce68804cb4
go: downloading github.com/cncf/udpa/go v0.0.0-20220112060539-c52dc94e7fbe
go: downloading github.com/envoyproxy/protoc-gen-validate v0.10.1
go: downloading github.com/census-instrumentation/opencensus-proto v0.4.1
# github.com/hashicorp/terraform-provider-google/google/services/compute
google/services/compute/resource_compute_instance_group_manager.go:481:3: unknown field InstanceLifecyclePolicy in struct literal of type "google.golang.org/api/compute/v1".InstanceGroupManager
google/services/compute/resource_compute_instance_group_manager.go:693:85: manager.InstanceLifecyclePolicy undefined (type *"google.golang.org/api/compute/v1".InstanceGroupManager has no field or method InstanceLifecyclePolicy)
google/services/compute/resource_compute_instance_group_manager.go:762:18: updatedManager.InstanceLifecyclePolicy undefined (type *"google.golang.org/api/compute/v1".InstanceGroupManager has no field or method InstanceLifecyclePolicy)
google/services/compute/resource_compute_instance_group_manager.go:1033:71: undefined: compute.InstanceGroupManagerInstanceLifecyclePolicy
google/services/compute/resource_compute_instance_group_manager.go:1034:38: undefined: compute.InstanceGroupManagerInstanceLifecyclePolicy
google/services/compute/resource_compute_instance_group_manager.go:1147:70: undefined: compute.InstanceGroupManagerInstanceLifecyclePolicy
google/services/compute/resource_compute_region_instance_group_manager.go:455:3: unknown field InstanceLifecyclePolicy in struct literal of type "google.golang.org/api/compute/v1".InstanceGroupManager
google/services/compute/resource_compute_region_instance_group_manager.go:639:85: manager.InstanceLifecyclePolicy undefined (type *"google.golang.org/api/compute/v1".InstanceGroupManager has no field or method InstanceLifecyclePolicy)
google/services/compute/resource_compute_region_instance_group_manager.go:704:18: updatedManager.InstanceLifecyclePolicy undefined (type *"google.golang.org/api/compute/v1".InstanceGroupManager has no field or method InstanceLifecyclePolicy)
make: *** [vet] Error 1
```

**Comment by askubis** [comment]
Oh, thanks!
I pulled and it seems resolved now.

**Review by NickElliot — COMMENTED** [review]
Could you share the log of it building successfully on your machine? It is still failing in both CI and for me locally and want to see if there is some unexpected difference happening here.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (NickElliot) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:1]:
```suggestion
```

**Review by NickElliot — COMMENTED** [review]
LGTM! Will approve to merge once this one accidental line of whitespace that got added is removed.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown (hunk 1 above)
Comment (askubis) [reply mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown]: I removed it, thanks!

**Review by NickElliot — APPROVED** [review]

**Pull request merged**